src/Language/PureScript/TypeChecker/Kinds.hs

```diff
@@ -419,6 +419,7 @@ unifyKindsWithFailure
   -> m ()
 unifyKindsWithFailure onFailure = go
   where
+  goWithLabel l t1 t2 = withErrorMessageHint (ErrorInRowLabel l) $ go t1 t2
```

Comment:
Instead of defining these local helpers in three different places, could you add a helper next to `alignRowsWith` called

```hs
alignRowsWithHinted
  :: (MonadState CheckState m, MonadError MultipleErrors m)
  => (Type a -> Type a -> m ())
  -> Type a -> Type a -> (m (), (([RowListItem a], Type a), ([RowListItem a], Type a)))
```
(note the absence of `Label` in this type), which simply delegates to your updated `alignRowsWith` and adds the hint? (I'd also factor the `sequence_` call that co-occurs with all of these call sites into this helper, which is why `m ()` and not `[m ()]`.)

Reply:
I've finally had enough time to look at this. But this is not really possible to do. We get a dependency cycle:
```
Module imports form a cycle:
        module ‘Language.PureScript.Types’ (src/Language/PureScript/Types.hs)
        imports module ‘Language.PureScript.Errors’ (src/Language/PureScript/Errors.hs)
  which imports module ‘Language.PureScript.Types’ (src/Language/PureScript/Types.hs)
```

I'm not sure I want to touch this much code and if this is the better change. It would have been great if it worked though. I'm all for exposing less and not moving complexity to the call sites. Curse your cyclic dependencies!

Reply:
Oh, that's unfortunate. Fine to leave as-is, then.